### PR TITLE
Feature/maintenance mode

### PR DESCRIPTION
**Pull request by markspl**
- Create maintenance view when env variable `REACT_APP_MAINTENANCE_MODE` is changed
- Tests for the MaintenanceView component
- Disable buttons and search buttons if the mode is set true

Env variables are already added to pipelines as 'false'.

### CODE REVIEW

**Review by millaeerola — APPROVED** [review]
Looks good! 👍

File: src/components/AuthGuard/AuthGuard.tsx [hunk 1]
```diff
@@ -165,6 +168,29 @@ const AuthGuard: FC = () => {
...
+    if (MAINTENANCE_MODE) {
+      if (pathname.includes(PAGES.ACCESS_DENIED) || pathname.includes(PAGES.AUTH_HELSINKI_RETURN)) {
+        return;
+      }
+
+      if (!pathname.includes(PAGES.MAINTENANCE_MODE)) {
+        return navigate(PAGES.MAINTENANCE_MODE);
+      }
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (millaeerola) [reply src/components/AuthGuard/AuthGuard.tsx:184]: The first if is about the log in. So if the user isn't logged in or doesn't have access or the login is still in progress, we don't want to redirect the user anywhere yet. The second if is for redirecting the user to maintenance mode page if they try to access something else while maintenance mode is on. I think without that, the user would never be redirected to maintenance mode page while it's true.

[the remark this answers is not in the document]
Comment (millaeerola) [reply src/components/AuthGuard/AuthGuard.tsx:184]: But the first one could maybe be deleted cause there's a return after the second if 🤔

[the remark this answers is not in the document]
Comment (millaeerola) [reply src/components/AuthGuard/AuthGuard.tsx:184]: Or actually not, cause the login thing needs to be checked first. Sorry if I'm causing just more confusion! 😄

**Review by JenniLehtonen — APPROVED** [review]

File: src/components/AuthGuard/AuthGuard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (markspl) [reply src/components/AuthGuard/AuthGuard.tsx:184]: Yeah, the row 176 is needed so redirects are finished correctly. There is a change the page gets stuck into auth-maintenance loop.

**Pull request merged**